src/libraries/System.Private.CoreLib/src/System/IO/StreamWriter.cs

```diff
@@ -224,12 +228,12 @@ private void CloseStreamFromDispose(bool disposing)
                         _stream.Close();
                     }
                 }
-                finally
-                {
-                    _disposed = true;
-                    _charLen = 0;
-                    base.Dispose(disposing);
-                }
+            }
+            finally
```

Comment:
What happens if _closable = false? Is there a change in behavior?

Reply:
Why did you move the condition inside the `try` block? If `_closable` is false, the stream shouldn't be disposed.

Reply:
I don't think so, it looks like `base.Dispose(bool disposing)` doesn't do anything.

Reply:
Before, _diposed and _charLen were affected only when the Stream was _closable, now it happens regardless.

Reply:
But isn't that correct? The `Stream` wasn't disposed, but the `StreamWriter` itself was.

Reply:
it's right, that _disposed and _charLen are now reset, but this is what dispose should do

Reply:
Well turned out, that there is a Test, that checks if it can continue writing to the underlying stream after calling Dispose on StreamWriter. So I reveted this change.
I also added the same test for DisposeAsync

Reply:
Srsly? 😅 Isn't calling anything after calling `Dispose` supposed to be undefined/disallowed? 🤦‍♀️

Reply:
@Neme12 I see what you mean now, so this behavior of "only set _dispose = true if _closable = true" is wrong but we now depend on it. @dersia what was the unit test that failed due to your change?

Reply:
it's this one

```cs
        [Theory]
        [MemberData(nameof(NullWriters))]
        public static void TextNullTextWriter(TextWriter output)
        {
            output.Flush();
            output.Dispose();

            output.WriteLine(decimal.MinValue);
            output.WriteLine(Math.PI);
            output.WriteLine();
            output.Flush();
            output.Dispose();
        }
```